File: Python/dominator.py
def dominator(arr):
    lenList = len(arr)
    newSetList = list(set(arr))
    currentDominator = { "key": 0, "value": 0}
    
    for number in newSetList:
        if arr.count(number) > currentDominator["value"]:
            currentDominator["key"] = number
            currentDominator["value"] = arr.count(number)

    if currentDominator["value"] > (lenList / 2):
        return currentDominator["key"]
    else:
        return -1
        

# print(dominator([3, 4, 3, 2, 3, 1, 3, 3]))
# print(dominator([1, 1, 1, 2, 2, 2, 2]))
# print(dominator([1, 2, 3, 4, 5]))
# print(dominator([1, 1, 1, 2, 2, 2]))

# print(dominator([6, 5, 10, 5, 5, 5, 5, 3, 5, 5, 10, 5]))
print(dominator([4, 4, 4, 4, 4, 4, 8]))

File: JavaScript/dominator.js
function dominator(arr) {
  const lenArr = arr.length;
  const dataObj = {};
  let count = 0;

  while (count < lenArr) {
    if (!dataObj[arr[count]]) {
      dataObj[arr[count]] = 0;
    }
    dataObj[arr[count]] += 1;
    count += 1
  }

  const dominator = { key: 0, value: 0 }

  for (let key in dataObj) {
    if (dataObj[key] > dominator.value) {
      dominator.key = +key
      dominator.value = dataObj[key]
    }
  }

  if (dominator.value > (lenArr / 2)) {
    return dominator.key;
  } else {
    return -1
  }
}


console.assert(dominator([3, 4, 3, 2, 3, 1, 3, 3]) === 3, "erro no teste 1")
// console.assert(dominator([1, 1 ,1 ,2 ,2 ,2 ,2]) === 2, "erro no teste 2")
// console.assert(dominator([1, 2, 3, 4, 5]) === -1, "erro no teste 3")
// console.assert(dominator([1, 1, 1, 2, 2, 2]) === -1, "erro no teste 4")